File: templates/Uwp/_composition/MVVMLight/Page.MediaPlayer.AddViewModelVB/ViewModels/ViewModelLocator_postaction.vb
Namespace ViewModels
    Public Class ViewModelLocator
        Private Sub New()
            '^^
            '{[{
            Register(Of wts.ItemNameViewModel, wts.ItemNamePage)()
            '}]}
        End Sub

'{[{

        Public ReadOnly Property wts.ItemNameViewModel As wts.ItemNameViewModel
            Get
                ' A Guid is generated as a unique key for each instance as reusing the same VM instance in multiple MediaPlayerElement instances can cause playback errors
                Return SimpleIoc.[Default].GetInstance(Of wts.ItemNameViewModel)(Guid.NewGuid().ToString())
            End Get
        End Property
'}]}
    End Class
End Namespace

File: templates/Uwp/_composition/MVVMLight/Page.MediaPlayer.AddViewModel/ViewModels/ViewModelLocator_postaction.cs
namespace Param_RootNamespace.ViewModels
{
    public class ViewModelLocator
    {
        private ViewModelLocator()
        {
            //^^
            //{[{
            Register<wts.ItemNameViewModel, wts.ItemNamePage>();
            //}]}
        }

        //{[{
        // A Guid is generated as a unique key for each instance as reusing the same VM instance in multiple MediaPlayerElement instances can cause playback errors
        public wts.ItemNameViewModel wts.ItemNameViewModel => SimpleIoc.Default.GetInstance<wts.ItemNameViewModel>(Guid.NewGuid().ToString());
        //}]}
    }
}
